src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala

```diff
@@ -95,50 +100,68 @@ trait UserService extends HttpService with PerRequestCreator with FireCloudReque
                         } else {
                           // rawls knows about the user, but the user isn't activated
-                          respondWithUserInfo(Forbidden, "FireCloud user not activated", requestContext)
+                          respondWithErrorReport(Forbidden, "FireCloud user not activated", requestContext)
                         }
                       // we couldn't parse the rawls response. Respond with an error.
                       case Left(error) =>
-                        respondWithUserInfo(InternalServerError, InternalServerError.defaultMessage, requestContext)
+                        respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
                     }
                   case x =>
                     // we got an unexpected error code from rawls; pass it on
-                    respondWithUserInfo(InternalServerError, "Unexpected response validating registration: " + x.toString, requestContext)
+                    respondWithErrorReport(InternalServerError, "Unexpected response validating registration: " + x.toString, requestContext)
                 }
               // we couldn't reach rawls (within timeout period). Respond with an error.
               case Failure(error) =>
-                respondWithUserInfo(InternalServerError, InternalServerError.defaultMessage, requestContext)
+                respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
             }
         }
       }
     } ~
     requireUserInfo() { userInfo =>
     pathPrefix("api") {
-      path("profile" / "billing") { requestContext =>
-        val extReq = Get(UserService.billingUrl)
-        externalHttpPerRequest(requestContext, extReq)
+      path("profile" / "billing") {
+        passthrough(UserService.billingUrl, HttpMethods.GET)
       } ~
-      path("profile" / "refreshTokenDate") { requestContext =>
-        val extReq = Get(OAuthService.remoteTokenDateUrl)
-        externalHttpPerRequest(requestContext, extReq)
+      path("profile" / "billingAccounts") {
+        oauthParams { (state, _) => requestContext =>
+          val pipeline = authHeaders(requestContext) ~> sendReceive
+          val extReq = Get(UserService.billingAccountsUrl)
+          pipeline(extReq) onComplete {
+            case Success(response) if response.status == Forbidden =>
+              val tryParseScopes = Try {
+                response.entity.asString.parseJson.convertTo[ErrorReport].message.parseJson.convertTo[BillingAccountScopes]
+              }
+              tryParseScopes match {
+                case Success(scopes) =>
+                  // user does not have appropriate scopes.  Ask user to enable them via OAuth
+                  val redirectURI = HttpGoogleServicesDAO.getGoogleRedirectURI(state, "auto", Option(scopes.requiredScopes ++ HttpGoogleServicesDAO.userLoginScopes))
```

Comment:
I *think* you can do this by redirecting to `[url]?scope=.https://www.googleapis.com/auth/cloud-billing.readonly&prompt=consent&include_granted_scopes=true`, instead of using the standard flow and requesting all scopes again. Of course this would require a few changes elsewhere, such as handling the `consent` value for the prompt.

Search for "Incremental authorization" on this page: https://developers.google.com/identity/protocols/OAuth2WebServer